### PR TITLE
Allow explicit spread of props

### PR DESCRIPTION
**Pull request by pawelnvk**
PR to resolve: https://github.com/yannickcr/eslint-plugin-react/issues/2439

### CODE REVIEW

File: docs/rules/jsx-props-no-spreading.md [hunk 1]
```diff
@@ -27,8 +27,9 @@ const {one_prop, two_prop} = otherProps;
 ```js
 ...
 "react/jsx-props-no-spreading": [{
-    "html": "ignore" / "enforce",
-    "custom": "ignore" / "enforce",
+    "html": "ignore"|"enforce",
```
Comment (ljharb) [inline docs/rules/jsx-props-no-spreading.md]:
```suggestion
    "html": "ignore" | "enforce",
```

File: docs/rules/jsx-props-no-spreading.md [hunk 2]
```diff
@@ -27,8 +27,9 @@ const {one_prop, two_prop} = otherProps;
...
+    "custom": "ignore"|"enforce",
```
Comment (ljharb) [inline docs/rules/jsx-props-no-spreading.md]:
```suggestion
    "custom": "ignore" | "enforce",
```

File: docs/rules/jsx-props-no-spreading.md [hunk 3]
```diff
@@ -27,8 +27,9 @@ const {one_prop, two_prop} = otherProps;
...
+    "explicitSpread": "ignore"|"enforce",
```
Comment (ljharb) [inline docs/rules/jsx-props-no-spreading.md]:
```suggestion
    "explicitSpread": "ignore" | "enforce",
```

File: lib/rules/jsx-props-no-spreading.js [hunk 4]
```diff
@@ -86,6 +93,11 @@ module.exports = {
           (!ignoreCustomTags && isException(tagName, exceptions)))) {
           return;
         }
+        if (ignoreExplicitSpread &&
+          node.argument.type === 'ObjectExpression' &&
+          node.argument.properties.every(isProperty)) {
```
Comment (ljharb) [inline lib/rules/jsx-props-no-spreading.js]:
```suggestion
          node.argument.properties.every(isProperty)
        ) {
```

File: lib/rules/jsx-props-no-spreading.js [hunk 5]
```diff
@@ -86,6 +93,11 @@ module.exports = {
           (!ignoreCustomTags && isException(tagName, exceptions)))) {
           return;
         }
+        if (ignoreExplicitSpread &&
```
Comment (ljharb) [inline lib/rules/jsx-props-no-spreading.js]:
```suggestion
        if (
          ignoreExplicitSpread &&
```

**Comment by pawelnvk** [comment]
@ljharb What do you think of prettier? Perhaps it could help us to keep consistent formatting across whole library.

**Comment by ljharb** [comment]
@pawelnvk no thanks; that’s what eslint is for.

**Pull request merged**